Support further / mismatched block-restricted variables in block-restricted sampled output

GiudGiud opened the pull request:
refs  #31114

GiudGiud committed:
Remove variables that got block-restricted out from block-restricted sampled output
refs  #31114

GiudGiud committed:
Use the source variable block restriction for sampling to enable sampling with overlapping meshes

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3078457/), step Docs: sync website on 178fd3a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31159/site)

This comment will be updated on new commits.

GiudGiud committed:
Do not use block restriction for sampling when the variable is not block restricted
Empty block restriction for 'no-block-restriction' is not understood that way by the mesh function

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3088910/), step Generate coverage on ce78b37 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>13e7aa</th>
      <th colspan=4>#31159 ce78b3</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.55%</td>
      <td>85.54%</td>
      <td>-0.01%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>116046</td>
      <td>116052</td>
      <td>+6</td>
      <td>25</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19605</td>
      <td>19617</td>
      <td>+12</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31159/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31159/coverage/framework)


## Modules coverage

Inconsistent report tags were found between the head and base reports.
This can happen when reports are missing from either the head or the base.

Inconsistent tags:
  module-chemical_reactions-restep
  module-contact-restep
  module-heat_transfer-restep

### Fluid properties

<table>
  <thead>
    <tr>
      <th></th>
      <th>13e7aa</th>
      <th colspan=4>#31159 ce78b3</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>86.01%</td>
      <td>86.00%</td>
      <td>-0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>8180</td>
      <td>8179</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1331</td>
      <td>1332</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31159/coverage_diff/fluid_properties)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31159/coverage/fluid_properties)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31159/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
I dont think so since it s the same failures as on this PR?
https://github.com/idaholab/moose/pull/31189

moosebuild commented:
Job [Controlled apps](https://civet.inl.gov/job/3088894/) on ce78b37 : invalidated by @GiudGiud

I think unrelated but let s be sure

GiudGiud commented:
Restep failure unrelated
